libs/dextool/source/dextool/compilation_db/package.d

```diff
@@ -701,6 +703,17 @@ ParseFlags parseFlag(CompileCommand cmd, const CompileCommandFilter flag_filter,
         return 0 != flag.among("-D");
     }
 
+    /// Flags that is includes but contains spaces are wrapped in quotation marks.
+    static bool isSpacedInclude(string flag) @safe {
+	return flag[0 .. 2] == "-I" && indexOf(flag, '"') == 2;
+    }
+
+    /// When we know we are building a path that is space separated,
+    /// the last index of the last string should be a quotation mark.
+    static bool isEndingWithQuotation(string flag) @safe {
+	return indexOf(flag, '"') == (flag.length-1);
```

Comment:
Another way of doing it is `!flag.empty && flag[$-1] == '"'`. The `$` is the last index.